src/Microsoft.OData.Core/WriterValidator.cs

```diff
@@ -20,24 +20,45 @@ internal class WriterValidator : IWriterValidator
         /// </summary>
         private readonly ODataMessageWriterSettings settings;
 
+        /// <summary>
+        /// Object pool that stores instances of the IDuplicatePropertyNameChecker.
+        /// </summary>
+        private ObjectPool<IDuplicatePropertyNameChecker> duplicatePropertyNameCheckerObjectPool;
+
         /// <summary>
         /// Creates a WriterValidator instance and binds it to settings.
         /// </summary>
         /// <param name="settings">The ODataMessageWriterSettings instance to bind to.</param>
         internal WriterValidator(ODataMessageWriterSettings settings)
         {
             this.settings = settings;
+            this.duplicatePropertyNameCheckerObjectPool = settings.ThrowOnDuplicatePropertyNames ?
+                new ObjectPool<IDuplicatePropertyNameChecker>(() => new DuplicatePropertyNameChecker()) :
+                new ObjectPool<IDuplicatePropertyNameChecker>(() => new NullDuplicatePropertyNameChecker());

```

Comment:
>NullDuplicatePropertyNameChecker [](http://example.com/codeflow?start=72&length=32)

shall we pool the NullDuplicatePropertyNameChecker

It seems we can use 'ONE' null duplicatePropertyNameChecker?

Reply:
#Resolved